### PR TITLE
Bump JOGL 2.4 to the new release candidate (20200307)

### PR DESCRIPTION
**Pull request by sampottinger**
Related to #84. Looks like RC 20200307 is promising! Note that this is blocked on #74 and #76. Also, note that there are no user-facing warning messages on Mac. Living the dream! Will test this on other platforms shortly.

Screen Shot 2020-04-12 at 11 13 11 AM

### CODE REVIEW

**Comment by sampottinger** [comment]
Looking good on Linux.
Screen Shot 2020-04-12 at 11 23 27 AM

**Comment by sampottinger** [comment]
and windows!

Screen Shot 2020-04-12 at 11 31 26 AM

**Comment by hx2A** [comment]
Hi- can you tell me what version of JOGL is used by the latest version of Processing 3.x? I can't seem to figure it out from looking at that repo. Thanks!

**Comment by sampottinger** [comment]
Hey @hx2A! I suspect it was 2.3.2.

**Comment by hx2A** [comment]
Thanks @sampottinger! After upgrading to Java 11, I was getting some illegal access warning from inside the JOGL jars and upgrading to the JOGL 2.4 jars made it go away. Do you see any risks to using the JOGL 2.4 jars with Processing 3.4? Using the Processing4 repo also solves the problem. Which do you think is the more stable and reliable choice?

**Comment by sampottinger** [comment]
Hey there! There's a few PRs outstanding (with inter-dependencies) and the build on master has been broken on Mac for some time. So, I have an integration branch over at https://github.com/sampottinger/processing4 master. That branch includes this PR. This allows one to use Processing 4 again on a Mac along with all of the recent bug fixes. For your convenience, I have made builds available at https://www.datadrivenempathy.com/processing.

**Comment by hx2A** [comment]
The integration build and repo are both very helpful for my project. Thank you!

File: core/src/processing/opengl/PSurfaceJOGL.java [hunk 1]
```diff
@@ -255,22 +255,6 @@ protected void initDisplay() {
 
     GraphicsConfiguration config = awtDisplayDevice.getDefaultConfiguration();
     displayRect = config.getBounds();
-
-    /** See explanation at {@link PGL#fboAllowed} in PGL */
-    if (PApplet.platform == PConstants.MACOS) {
```
Comment (benfry) [inline core/src/processing/opengl/PSurfaceJOGL.java]: Why is this code being rewritten? It's a workaround that we want to remove as soon as possible, not something to be refactored and expanded.

**Comment by benfry** [comment]
Looks like several non-JOGL things have crept into this PR?

File: core/src/processing/opengl/PSurfaceJOGL.java (hunk 1 above)
Comment (sampottinger) [reply core/src/processing/opengl/PSurfaceJOGL.java]: Hey @benfry! This is actually from https://github.com/processing/processing4/pull/76. Sorry I didn't make it clear that this is blocked on / dependent on #76

**Comment by sampottinger** [comment]
Sorry @benfry this is blocked on #76. This is a branch off of that PR. I'll follow up in email in a minute with more details.

**Comment by sampottinger** [comment]
Just for folks following along, I'm adding an outline of how things are interconnected over at #74 .

**Comment by sampottinger** [comment]
Looks like this got superseded by changes already merged.

**Pull request merged**